lib/vm/src/vmoffsets.rs

```diff
@@ -479,13 +479,18 @@ impl VMOffsets {
             .unwrap()
     }
 
+    /// The offset of the stack limit.
+    pub fn vmctx_stack_limit_pointer(&self) -> u32 {
+        self.vmctx_gas_limiter_pointer()
+            .checked_add(u32::from(self.pointer_size))
+            .unwrap()
+    }
+
     /// Return the size of the [`VMContext`] allocation.
     ///
     /// [`VMContext`]: crate::vmcontext::VMContext
     pub fn size_of_vmctx(&self) -> u32 {
-        self.vmctx_gas_limiter_pointer()
-            .checked_add(u32::from(self.pointer_size))
-            .unwrap()
+        self.vmctx_stack_limit_pointer().checked_add(4).unwrap()
```

Comment:
why did we replace `self.pointer_size` with 4?

Reply:
The stack limit “pointer” is a `u32` value, rather than a proper pointer that gas limiter pointer was. This field should be renamed ^^

Reply:
I'll adjust the names to make it more clear.

Reply:
>I'll adjust the names to make it more clear.

I'd say the name is right. It's rather that `gas_counter_ptr` should be named `gas_counter_ptr_ptr`